### PR TITLE
add crypto on target testing script

### PR DESCRIPTION
**Pull request by netanelgonen**
### Description
WIP - adding a script to generate, compile and run the PSA crypto on mbed-os
do not review yet

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

---
This change is [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/85)

### CODE REVIEW

File: TESTS/psa/entropy_inject/main.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@
 * limitations under the License.
 */
 
-#if ((!defined(TARGET_PSA) || (!defined(COMPONENT_PSA_SRV_IPC)) && !defined(MBEDTLS_ENTROPY_NV_SEED)))
+#if (!defined(TARGET_PSA) || ((!defined(COMPONENT_PSA_SRV_IPC)) && !defined(MBEDTLS_ENTROPY_NV_SEED)))
```
Comment (Patater) [inline TESTS/psa/entropy_inject/main.cpp]: Where do we want the entropy injection tests to run? I'm not sure these defines are right if we want it to run on both K64F and on a PSoC 6's NSPE.

File: TESTS/psa/prot_internal_storage/COMPONENT_SPE/psa_setup.c [hunk 2]
```diff
@@ -46,14 +46,14 @@ spm_partition_t g_partitions[3] = {
         .irq_mapper = NULL,
     },
     {
-        .partition_id = PSA_F_ID,
+        .partition_id = CRYPTO_ID,
```
Comment (Patater) [inline TESTS/psa/prot_internal_storage/COMPONENT_SPE/psa_setup.c]: Do we want to change PSA_F_ID to CRYPTO_ID everywhere, even in non-crypto tests? Maybe the storage tests should use a storage partition and not a crypto partition?

File: tools/test_crypto.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+    return parser
+
+
+def psa_deploy():
+    logger.info("--- Set up psa-crypto env ---")
+
+    check_call(["make", "update"], cwd=MBEDTLS_IMPORTER_DIRECTORY)
+    crypto_repo = git.Repo(MBED_CRYPTO_DIRECTORY)
+    crypto_repo.git.checkout("feature-psa")
```
Comment (Patater) [inline tools/test_crypto.py]: If you care about only Mbed Crypto (and not any TLS APIs), this is OK. If you want to import TLS, you should use the development-psa branch of ARMmbed/mbedtls, which also pulls in Mbed Crypto through a submodule.

File: TESTS/psa/crypto_init/main.cpp [hunk 4]
```diff
@@ -16,11 +16,7 @@
 * limitations under the License.
 */
 
-<<<<<<< HEAD
```
Comment (Patater) [inline TESTS/psa/crypto_init/main.cpp]: Where did these conflict markers come from? If they were added in this PR, could we remove them in the original naughty commit that added them?

File: components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c [hunk 5]
```diff
@@ -208,647 +207,6 @@ static const osMutexAttr_t crypto_mutex_attr = {
...
-            break;
-
-        default:
-            status = PSA_ERROR_NOT_SUPPORTED;
-    }
-    psa_reply(msg.handle, (psa_error_t) status);
-}
-
-static void psa_symmetric_operation(void)
->>>>>>> Fix the crappy crypto code
```
Comment (Patater) [inline components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c]: Why don't you want any changes from "Fix the crappy crypto code"? It sounds like a commit that contains useful changes!

File: tools/test_crypto.py (hunk 3 above)
Comment (netanelgonen) [reply tools/test_crypto.py]: @Patater This is only for mbed crypto, but I need to add an option to choose the branch of mbed-crypto
if I will add an option to choose the branch of mbedtls and default crypto branch will it be OK?

**Comment by netanelgonen** [comment]
replacing this PR with a new one to mbed-os master
https://github.com/ARMmbed/mbed-os/pull/8976

**Pull request closed without merging**